### DISCUSSION TITLE
Version 7.42.0 Breaks Controller Component

### DISCUSSION THREAD

**Discussion by bjrice88** [react-hook-form/react-hook-form#9786] [Q&A]
Starting with Version 7.42.0 I am seeing a problem where some forms are giving an error of `TS2589: Type instantiation is excessively deep and possibly infinite.`  

There doesn't appear to be any rhyme or reason to where this happens, in my CI/CD pipeline it happens in some files and on my machine it happens in another but if it is TS-ignored it doesn't come up in the file that has an error in the CI pipeline.

Any Ideas?

**Comment by leapful** [comment]
@bjrice88 

Please provide a codesandbox or github repo to re-produce the issue.
